publication-rest/src/test/java/no/unit/nva/publication/delete/DeletePublicationHandlerTest.java

```diff
@@ -208,27 +187,333 @@ void handleRequestReturnsUnauthorizedWhenCallerIsMissingClientId()
+                              .build();
+
+        handler.handleRequest(inputStream, outputStream, context);
+
+        var response = GatewayResponse.fromOutputStream(outputStream, Void.class);
+        assertThat(response.getStatusCode(), is(equalTo(SC_UNAUTHORIZED)));
+    }
+
+    @Test
+    void shouldReturnSuccessWhenCuratorUnpublishesPublishedPublicationFromOwnInstitution()
+        throws ApiGatewayException, IOException {
+
+        var institutionId = randomUri();
+        var curatorUsername = randomString();
+        var resourceOwnerUsername = randomString();
+
+        var publication = createAndPersistPublicationWithoutDoiAndWithResourceOwner(resourceOwnerUsername,
+                                                                                    institutionId);
+        publicationService.publishPublication(UserInstance.fromPublication(publication), publication.getIdentifier());
+
+        var inputStream = new HandlerRequestBuilder<Void>(restApiMapper)
+                              .withUserName(curatorUsername)
+                              .withCurrentCustomer(institutionId)
+                              .withAccessRights(institutionId, AccessRight.EDIT_OWN_INSTITUTION_RESOURCES.name())
+                              .withPathParameters(
+                                  Map.of(PUBLICATION_IDENTIFIER, publication.getIdentifier().toString()))
+                              .build();
+
+        handler.handleRequest(inputStream, outputStream, context);
+
+        var response = GatewayResponse.fromOutputStream(outputStream, Void.class);
+        assertThat(response.getStatusCode(), is(equalTo(SC_ACCEPTED)));
+    }
+
+    @Test
+    void shouldReturnUnauthorizedWhenCuratorUnpublishesPublishedPublicationFromAnotherInstitution()
+        throws IOException, ApiGatewayException {
+
+        var curatorUsername = randomString();
+        var curatorInstitutionId = randomUri();
+
+        var resourceOwnerUsername = randomString();
+        var resourceOwnerInstitutionId = randomUri();
+
+        var publication = createAndPersistPublicationWithoutDoiAndWithResourceOwner(resourceOwnerUsername,
+                                                                                    resourceOwnerInstitutionId);
+        publicationService.publishPublication(UserInstance.fromPublication(publication), publication.getIdentifier());
+
+        var inputStream = new HandlerRequestBuilder<Void>(restApiMapper)
+                              .withUserName(curatorUsername)
+                              .withCurrentCustomer(curatorInstitutionId)
+                              .withAccessRights(curatorInstitutionId, AccessRight.EDIT_OWN_INSTITUTION_RESOURCES.name())
+                              .withPathParameters(
+                                  Map.of(PUBLICATION_IDENTIFIER, publication.getIdentifier().toString()))
+                              .build();
```

Comment:
extract to new method in all tests, for example requestWithAccessRight(accessRight)